style_guides/dart.md

```diff
@@ -0,0 +1,63 @@
+# Shorebird Dart Style Guide
+
+This document outlines the style guide for all Shorebird Dart code.
+
+This guide is a supplement to
+[Effective Dart](https://dart.dev/guides/language/effective-dart) and should be
+read in the same way. If there is a conflict between this guide and Effective
+Dart, this guide takes precedence.
+
+## AVOID mixins
+
+Mixins are a powerful feature of Dart, but they can make code harder to
+test. Mixins are effectively unmockable dependencies, and any code using them
+will need to fully mock out all of the mixin's dependencies. Prefer creating a
+new class as a `scoped` dependency instead.
+
+## DO define one variable per line
+
+```dart
+// BAD
```

Comment:
```suggestion
// AVOID
```

Reply:
This follows the Effective Dart layout (I'm not sure if there's a way to stylize code blocks as "good" or "bad" in GitHub markdown), so I'd prefer to keep GOOD and BAD unless you feel strongly.

Reply:
See https://dart.dev/effective-dart/documentation for what I mean

Reply:
Sounds good. I don't feel strongly just seemed like "GOOD" and "BAD" might be a bit harsh but happy to keep it consistent with effective dart 👍